reearth/reearth
# feat(web,server): support multi-tenant login

# Changed

- Add `multitenant` config to reearth_config.json
- Add `/auth/<tenant name>` page to login with an other tenant
- Make `team` field in `me` gql query optional (because if you are a guest, me is present but my workspace is not present)
- If there is no current workspace, project creation button should not bwork
- Refactor some workspace types
- Pretty-print config

# How to test

Both front-end and back-end should be set up:

- Auth0
  - Crate a new app and specify callback url: `<reearth origin>/auth/<tenant name>`
- front-end
   - Add auth info to reearth-config.json: `"multipletenant": { "<tenant name>": { "auth0Domain": "xxx", ... } }`
- back-end
   - Specify `REEARTH_DB_USERS` env vars: e.g. `<tennant name>=<DB URL>`
   - Specify `REEARTH_AUTH_ISS`: e.g. Auth0 domain
   - Specify `REEARTH_AUTH_AUD`: e.g. Auth0 audience

Then access `<reearth origin>/auth/<tenant name>` to login with the tenant.
